aquarium/aquarium.js

```diff
@@ -1517,8 +1521,12 @@ function initialize() {
           vertices[ii].push(fishInfo.heightRange);
         }
       });
-      gl.bindBuffer(gl.ARRAY_BUFFER, g_fishInfoBuffer);
-      gl.bufferData(gl.ARRAY_BUFFER, new Float32Array([].concat.apply([], vertices)), gl.STATIC_DRAW);
+      vertices = new Float32Array([].concat.apply([], vertices));
+      g_fishInfoBuffer = new tdl.buffers.Buffer({
```

Comment:
What's the motivation to do this? For OO style?

Reply:
To kill `getBufferParameter()`, the buffer size must be cached somewhere in some form. I don't like yet another ad-hoc global variable, and the `Buffer` object happens to fulfill the wish. To some extent, yes it's OO.

Reply:
Got it, thanks!